src/ArborX_LinearBVH.hpp

```diff
@@ -167,11 +183,20 @@ class BoundingVolumeHierarchy<
   {
   }
   // clang-format on
-  template <typename... Args>
-  void query(Args &&... args) const
+  template <typename FirstArgumentType, typename... Args>
+  std::enable_if_t<Kokkos::is_execution_space<FirstArgumentType>::value>
+  query(FirstArgumentType const &space, Args &&... args) const
   {
     BoundingVolumeHierarchy<typename DeviceType::memory_space>::query(
-        typename DeviceType::execution_space{}, std::forward<Args>(args)...);
+        space, std::forward<Args>(args)...);
+  }
+  template <typename FirstArgumentType, typename... Args>
+  std::enable_if_t<!Kokkos::is_execution_space<FirstArgumentType>::value>
+  query(FirstArgumentType &&arg1, Args &&... args) const
```

Comment:
You did not discuss this in the description of you PR.  I think it is beyond the scope of this PR.

Reply:
Daniel pointed out to me that I missed it was in the DeviceType specialization.

Reply:
In any case you are adding an overload

Reply:
To make things work with `ArborX::query`, `BVH<DeviceType>` has to be able to take in an execution space but do not attach another execution space to the call. Thus, I need a specialization for this. Therefore, there are two overloads, one to maintain backwards compatibility, one to use the new approach. I believe this falls squarely under this PR.

Reply:
I made the `query` that takes in an execution space private. This way, this PR does not affect the public interface of `BVH<DeviceType>`.